cilium/cmd/monitor.go

```diff
@@ -235,35 +240,46 @@ func runMonitor() {
 	signal.Notify(signalChan, os.Interrupt)
 	go func() {
 		for range signalChan {
-			fmt.Printf("\nReceived an interrupt, stopping monitor...\n\n")
-
-			lost, unknown := events.Stats()
-			if lost != 0 || unknown != 0 {
-				fmt.Printf("%d events lost, %d unknown notifications\n", lost, unknown)
-			}
-
-			if err := events.CloseAll(); err != nil {
-				panic(err)
-			}
-
+			fmt.Printf("\nReceived an interrupt, disconnecting from monitor...\n\n")
 			os.Exit(0)
 		}
 	}()
 
-	fmt.Printf("Listening for events on %d CPUs with %dx%d of shared memory\n",
-		events.Cpus, events.Npages, events.Pagesize)
-	fmt.Printf("Press Ctrl-C to quit\n")
-
+	var meta payload.Meta
+	metaBuf := make([]byte, unsafe.Sizeof(meta))
 	for {
-		todo, err := events.Poll(5000)
-		if err != nil && err != unix.EINTR {
-			panic(err)
+		// We need to know about the incoming payload otherwise can't know
+		// how much data to read.
+		if _, err := conn.Read(metaBuf); err != nil {
+			if err == io.EOF {
+				time.Sleep(12 * time.Second)
```

Comment:
In the reload make target, it takes normally 6 seconds. Observed that it sometimes takes more time for the daemon to restart. So doubled it.